From c4387612100f5740d6591c8c77f969cd05409dc8 Mon Sep 17 00:00:00 2001
From: serge-sans-paille <serge.guelton@telecom-bretagne.eu>
Date: Mon, 11 Dec 2023 10:40:19 +0100
Subject: [PATCH] Fix regression on XSIMD_NO_SUPPORTED_ARCHITECTURE introduced
 in e92cd8667231f25d88d233ceca04a64b063e33f3

Also add a test to make sure we don't regress in the future.

Fix #990
---
 third_party/xsimd/include/xsimd/config/xsimd_config.hpp | 1 +
 2 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/third_party/xsimd/include/xsimd/config/xsimd_config.hpp b/third_party/xsimd/include/xsimd/config/xsimd_config.hpp
index 31c4acb9e..ac416c70f 100644
--- a/third_party/xsimd/include/xsimd/config/xsimd_config.hpp
+++ b/third_party/xsimd/include/xsimd/config/xsimd_config.hpp
@@ -355,6 +355,7 @@
 #endif
 
 #if !XSIMD_WITH_SSE2 && !XSIMD_WITH_SSE3 && !XSIMD_WITH_SSSE3 && !XSIMD_WITH_SSE4_1 && !XSIMD_WITH_SSE4_2 && !XSIMD_WITH_AVX && !XSIMD_WITH_AVX2 && !XSIMD_WITH_FMA3_SSE && !XSIMD_WITH_FMA4 && !XSIMD_WITH_FMA3_AVX && !XSIMD_WITH_FMA3_AVX2 && !XSIMD_WITH_AVX512F && !XSIMD_WITH_AVX512CD && !XSIMD_WITH_AVX512DQ && !XSIMD_WITH_AVX512BW && !XSIMD_WITH_NEON && !XSIMD_WITH_NEON64 && !XSIMD_WITH_SVE && !XSIMD_WITH_WASM
+#define XSIMD_NO_SUPPORTED_ARCHITECTURE
 #endif
 
 #endif

